### DISCUSSION TITLE
Passing params from client signIn to signIn callback

### DISCUSSION THREAD

**Discussion by Amantel** [nextauthjs/next-auth#1363] [Help]
Hi! 
Imagine a simple situation.
You have a button like this: 

`<button onClick={() => signIn('spotify', { action: 'sing_in_no_register' })}>sing_in_no_register</button>`

When I click on it, as per function [description](https://github.com/nextauthjs/next-auth/blob/main/src/client/index.js#L245) I pass 'action' to the authorizationParams. 

There are two questions: 

1. Who I get that info in the `NextAuth` `callbacks` `signIn` function
2. Options are send as a part of GET query, thus, I can't send a lot of info there, correct?

Full disclosure: 
My goal is to do special mumbo-jumbo (like send additional API calls or open popus in UI) based on additional information. Of course, I can save everything to some kind of storage (cookie, localstorage), but maybe there is a better way. 

UPD. Using `he third argument of signIn().` bids same results

**Comment by balazsorban44** [comment]
so could you please specify special mumbo-jumbo? 😄 do you want to do it before the request to the oauth provider?

**Reply by Amantel** [reply]
Will do :)
Basically I have two sets of cases. 
First - front-end ones. Depending on the button clicked I want to display (in simple terms) some info.
Ex1: user clicks on 'register with spotify', I display his newly created profile in a popup. 
Ex2: user clicks on 'use spotify to show me my bands' and I display him a list of bands connected to his spotify account.

Second - 'back-endish'. 
in Ex1, after the request to oauth provider is completed, I would like to do an API query to the back-end which will tell return additional info about newly generated user.
in Ex2, after the request to oauth provider is completed, I would like to do additional call to spotify API to get user bands. Also I do not want to do that call in Ex1 (and vice versa)

**Reply by Amantel** [reply]
Found even better (artificial, though) example. 
As per tip in [https://next-auth.js.org/configuration/callbacks](url) I can block users that are not in database from authorizing, if they are not in the database. Now image the situation, where I have 2 buttons - login and registration. First one must behave exactly as described, e.g. returned 'you do not have an account with us' message and second must behave normally.

**Reply by Amantel** [reply]
And another one: 
I have a registration form where user types his city and surname, and obviously I want to put that info to the database (and access it in the session later). 

If previous questions where about **transactional variables**, this is more about updating the user model, so this case, probably should be handled by  custom models and may be not a part of the question.
